### PR TITLE
URL/Encoding: change query state parsing

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/encoding/issues/139 for rationale and https://github.com/whatwg/url/pull/386 for the change to the URL Standard.

(I found all these resources in part due to @rakuco's work on trying to align Chrome with the earlier iteration of the specification.)

### CODE REVIEW

**Comment by annevk** [comment]
Note: I looked into removing `normalizeStr` completely, but that broke too many tests. So I only removed it where it was a no-op or caused only one browser (Firefox) to start failing tests.

Perhaps further cleanup here can be done as part of #10636.

**Comment by annevk** [comment]
That's now done. This as well the URL Standard change are ready for review.

**Comment by annevk** [comment]
Note to self: once this is reviewed and landed, do some work on #4934 again.

**Review by inexorabletash — APPROVED** [review]
lgtm

**Comment by annevk** [comment]
(I merged the commits together so that "rebase and merge" can be used by an owner. This needs an override due to build timeouts. However, please do not merge this before I remove that label.)

**Comment by annevk** [comment]
Also, thanks @inexorabletash for the speedy review!

**Comment by annevk** [comment]
I rebased this to address a merge conflict in `url/README.md`.

**Pull request merged**
